packages/vm/src/cache.rs

```diff
@@ -117,7 +118,12 @@ pub struct Cache<A: BackendApi, S: Storage, Q: Querier> {
 
 #[derive(PartialEq, Eq, Debug)]
 pub struct AnalysisReport {
+    /// `true` if and only if all [`REQUIRED_IBC_EXPORTS`] exist as exported functions.
+    /// This does not guarantee they are functional or even have the correct signatures.
     pub has_ibc_entry_points: bool,
+    /// A set of all entrypoints that are exported by the contract.
+    pub entrypoints: HashSet<Entrypoint>,
```

Comment:
Would it make sense to use a BTreeSet instead? Or does this not work because it would not be sorted by string?  I'm just wondering if we can avoild this sorting step in wasmvm when turning the set to a comma separated sorted list:

```
fn set_to_csv(set: HashSet<String>) -> String {
    let mut list: Vec<String> = set.into_iter().collect();
    list.sort_unstable();
    list.join(",")
}
```

If this becomes complicated, just leave it as is. Then we sort in wasmvm, no big deal.

Reply:
Great point! I didn't notice that we manually sort in wasmvm. BTreeSet is perfect for this.

Reply:
For `required_capabilities`, I didn't change this in the whole codebase and just convert to `BTreeSet` in the `analyze` call.